### ISSUE TITLE
[Feature Addition]: Web App for Wine Quality Prediction

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#601]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : [Feature Addition]: Web App for Wine Quality Prediction
</br></br>:red_circle: **Aim** : The aim is to create a web app for the [Wine Quality Prediction](https://github.com/abhisheks008/ML-Crate/tree/main/Wine%20Quality%20Prediction) project. Use the best existing model in the project folder. Follow the [Web App README template](https://github.com/abhisheks008/ML-Crate/blob/main/.github/web-app-readme-template.md) for the same.
</br></br>:red_circle: **Dataset** : N/A
</br></br>:red_circle: **Approach** : Try to create a Flask/Streamlit app for the existing project.
</br></br>🔴 **Reference Project Folder**: [Brain Tumor Detection](https://github.com/abhisheks008/DL-Simplified/tree/main/Brain%20Tumor%20Detection)

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by shivansh-2003** [comment]
Can You Please Assign this issue under SSOC. 2024 Season 3
Shivansh Mahajan
Github:- https://github.com/shivansh-2003
Participation ID:- NA
My approach will be first data cleaning , feature engineering of the dataset . Then i will proceed toward ML models and try different Classification Model And Assign Quality score distribution . After That I will make a pickle file of the Fitted ML model
and then deploy it to Streamlit Python Framework to display the feature and predict the quality

**Comment by Anshg07** [comment]
Full name: Ansh Gupta
GitHub Profile Link: [Anshg07](https://github.com/Anshg07)
Participant ID: NA
Approach for this Project: I will develop a web app using Flask/Streamlit for the Wine Quality Prediction project. I will utilize the best existing model in the project folder. Following the Web App README template, I will create a separate folder named "Wine Quality Prediction" with subfolders for Images, Dataset, Model, and a requirements.txt file. The Model folder will include a README.md with visualizations and conclusions.

Participation Role : SSOC Season 3

**Comment by uchiha-vivek** [comment]
Full Name : Vivek Sharma
Github Link : github.com/uchiha-vivek
Participant ID :NA
Approach : 

Frontend: 

1-) I will make a Frontend app using React.js and first design a good looking Landing Page.(## including Header,Footer and complete responsive web app)
2-)Next step would be to make a predictor component based on features like fixed acidity,volatile acidity and different features present in Dataset in order to to predict the quality of wine.
3-) The whole predictor system for analyzing the quality of wine would be incorporated within a Form . 
Note : The form can be made in Landing Page or the web app could be made more functional according to instructions of admin.
4-)The inputs of form will render data in form of json from backend. 

Backend:

1-) The backend server can be made in flask and the whole trained model can be converted into pickel file.
2-) The routes will be adjusted accordingly.
3-) The existing model can also be retrained in order to increase the accuracy if possible.

Testing - The API's will be tested using Postman to check whether the data is rendered properly or not.
README.md files will be updated in parallel with progress of the project.
Also since the value for most of the features are discrete therefore in form we can round it off to 2 decimal places.

Experience : previously worked as student intern in immer(US startup) as full stack developer intern
Linkedin: www.linkedin.com/in/vivekuchiha
Sir can you please assign me this issue @abhisheks008 

Participation Role:SSOC Season 3

**Comment by abhisheks008** [comment]
Hi @uchiha-vivek use the existing model which is having the highest accuracy score for the web app.
Assigned to you.

**Comment by uchiha-vivek** [comment]
@abhisheks008  Ok sir i will use the existing model.

Thank you sir.
